File: Android/ViewsSwitchExample/app/src/main/java/com/example/viewsswitchexample/MainActivity.java
package com.example.viewsswitchexample;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        var aboutButton = findViewById(R.id.linkAboutButton);
        aboutButton.setOnClickListener(view -> {
            Intent intent = new Intent(getApplicationContext(), AboutActivity.class);
            startActivity(intent);
        });

        var inputDataButton = findViewById(R.id.inputDataButton);
        inputDataButton.setOnClickListener(view -> {
            Intent intent = new Intent(getApplicationContext(), InputDataActivity.class);
            startActivity(intent);
        });

        var inputData2Button = findViewById(R.id.inputData2Button);
        inputData2Button.setOnClickListener(view -> {
            Intent intent = new Intent(getApplicationContext(), SecondActivity.class);
            startActivityForResult(intent, 42);
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == 42) {
            if (resultCode == RESULT_OK) {
                var dataValue = data.getStringExtra("data");
                Toast.makeText(this, dataValue, Toast.LENGTH_SHORT).show();
            } else {
                Toast.makeText(this, "Request cancelled", Toast.LENGTH_SHORT).show();
            }
        }
    }
}

File: Android/AndroidPracticeThree/app/src/main/java/com/example/androidpracticethree/MainActivity.kt
package com.example.androidpracticethree

import android.app.AlertDialog
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var nameEditText: EditText
    private lateinit var descriptionEditText: EditText
    private lateinit var toDateEditText: EditText
    private var taskList: ArrayList<Task> = ArrayList();

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        nameEditText = findViewById(R.id.nameEditText)
        descriptionEditText = findViewById(R.id.descriptionEditText)
        toDateEditText = findViewById(R.id.toDateEditText)
    }

    fun onSubmitButtonClick(view: View) {
        if (toDateEditText.text.isEmpty() || descriptionEditText.text.isEmpty() || nameEditText.text.isEmpty()) {
            Toast.makeText(this, "Заполните все поля!", Toast.LENGTH_LONG).show()
            return;
        }
        val name = nameEditText.text.toString();
        val description = descriptionEditText.text.toString();
        val toDate = toDateEditText.text.toString();

        val newTask = Task(name, description, toDate);
        taskList.add(newTask);

        Toast.makeText(this, "Успешно добавлено!", Toast.LENGTH_LONG).show()
        nameEditText.text = null;
        descriptionEditText.text = null;
        toDateEditText.text = null;
    }

    fun onShowButtonClick(view: View) {
        val sb = StringBuilder()
        for (task in taskList) {
            sb.appendLine("${task.name} - ${task.description}; Выполнить до: ${task.toDate}")
        }

        val text = sb.toString();
        val builder = AlertDialog.Builder(this);
        builder.setTitle("Список дел").setMessage(text)
        val dialog = builder.create();
        dialog.show();
    }
}